oaknational/Oak-Web-Application
# fix: use isLoggedIn to prevent excess bookmark requests

## Description

**Note: CI will not pass until the Hasura PR goes in and gets merged in to this**

- Adds `isLoggedIn` as convenience prop to `useAuth` return value
- Use `isLoggedIn` as dependency list item, so that fetchBookarks only happens on log in or load.
  - Previously `fetchBookmarks` was called any time a new `oakUser` object was returned, which more frequent than necessary

## How to test

1. Go to {cloud link}, with network tab open
2. refresh page and/or log in
3. Should only see one "graphql" request
